stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/DateFormatter.kt

```diff
@@ -0,0 +1,20 @@
+package io.getstream.chat.android.ui.utils
+
+import android.content.Context
+import java.time.LocalDateTime
+import java.util.Date
+
+public interface DateFormatter {
+    public fun formatMessageDate(localDateTime: LocalDateTime?): String
+
+    public companion object {
+        public fun from(context: Context): DateFormatter = DefaultDateFormatter(context)
+    }
+}
+
+/**
+ * Extension to be able to format objects of the deprecated [Date] type.
+ */
+internal fun DateFormatter.formatMessageDate(date: Date?): String {
+    return formatMessageDate(date?.let(DateConverter::toLocalDateTime))
+}
```

Comment:
Didn't want to write a bunch of new Date handling code and tests built on the old Date APIs. This extension is a bridge from the old Date APIs to the new ones, so that the new formatter can be used with `java.util.Date` for the time being. Ideally, this will be removed at some point.